eth/tracers/api.go

```diff
@@ -740,7 +768,19 @@ func (api *API) standardTraceBlockToFile(ctx context.Context, block *types.Block
 		// Execute the transaction and flush any traces to disk
 		vmenv := vm.NewEVM(vmctx, txContext, statedb, chainConfig, vmConf)
 		statedb.Prepare(tx.Hash(), i)
-		_, err = core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas()))
+		l1DataFee, err := fees.CalculateL1DataFee(tx, statedb)
+		if err != nil {
+			if writer != nil {
+				writer.Flush()
+			}
+			if dump != nil {
+				dump.Close()
+				log.Info("Wrote standard trace due to CalculateL1DataFee error", "file", dump.Name(), "err", err)
+			}
+			return dumps, err
+		}
+
+		_, err = core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas()), l1DataFee)
```

Comment:
This could be cleaner. The `writer`, `dump`, and `err` check do not need to be duplicated.

```go
l1DataFee, err := fees.CalculateL1DataFee(tx, statedb)
if err == nil {
    _, err = core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas()), l1DataFee)
}
if writer != nil {
// ...
```

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/351/commits/2753504c3518df5f66ecad89caf7e967d02dbb4c